packages/itwin/reports-config-widget/src/test/ReportMappings.test.tsx

```diff
@@ -419,289 +384,40 @@ describe("Report Mappings View", () => {
-        self: {
-          href: "",
-        },
-      },
-    };
-
-    server.use(
-      rest.get(
-        `${REPORTS_CONFIG_BASE_URL}/insights/reporting/reports/${mockReportId}/datasources/imodelMappings`,
-        async (_req, res, ctx) => {
-          return res(
-            ctx.delay(),
-            ctx.status(200),
-            ctx.json(mockReportMappings)
-          );
-        }
-      ),
-      rest.get(`${Constants.api.baseUrl}`, async (_req, res, ctx) => {
-        return res(ctx.delay(), ctx.status(200), ctx.json(mockProjectIModels));
-      }),
-      rest.get(
-        `${Constants.api.baseUrl}/${mockIModelId1}`,
-        async (_req, res, ctx) => {
-          return res(
-            ctx.delay(),
-            ctx.status(200),
-            ctx.json(mockIModelsResponse[0])
-          );
-        }
-      ),
-      rest.get(
-        `${Constants.api.baseUrl}/${mockIModelId2}`,
-        async (_req, res, ctx) => {
-          return res(
-            ctx.delay(),
-            ctx.status(200),
-            ctx.json(mockIModelsResponse[1])
-          );
-        }
-      ),
-      rest.get(
-        `${REPORTS_CONFIG_BASE_URL}/insights/reporting/datasources/imodels/${mockProjectIModels.iModels[0].id}/mappings`,
-        async (_req, res, ctx) => {
-          return res(
-            ctx.delay(),
-            ctx.status(200),
-            ctx.json(mockMappingsResponse)
-          );
-        }
-      ),
-      rest.post(
-        `${REPORTS_CONFIG_BASE_URL}/insights/reporting/reports/${mockReportId}/datasources/imodelMappings`,
-        async (_req, res, ctx) => {
-          return res(
-            ctx.delay(),
-            ctx.status(200),
-            ctx.json(mockReportMappings)
-          );
-        }
-      ),
-      ...iModelHandlers
-    );
+    mockGetMapping.mockReturnValueOnce(mockMappings[0].mapping).mockReturnValueOnce(mockMappings[1].mapping);
+    mockGetReportMappings.mockReturnValueOnce(mockReportMappings.mappings);
 
-    const { user } = render(
-      <ReportMappings report={mockReport} goBack={jest.fn()} />
+    render(
+      <ReportMappings report={mockReport} bulkExtractor={mockBulkExtractor.object} goBack={jest.fn()} />
     );
 
     await waitForElementToBeRemoved(() => screen.getByText(/loading/i));
 
-    const addMappingButton = screen.getByRole("button", {
-      name: /addmapping/i,
-    });
-
-    await user.click(addMappingButton);
-
-    await waitForElementToBeRemoved(() =>
-      screen.getByTestId(/rcw-action-loading-spinner/i)
-    );
-    // Add modal dialog should appear
-    const modal = screen.getByRole("dialog");
-    expect(screen.getByRole("dialog")).toBeInTheDocument();
-
-    const withinModal = within(modal);
-    expect(withinModal.getByText(/addmappings/i)).toBeInTheDocument();
+    mockGetMappings.mockReturnValueOnce(mockMappings.map((m: MappingSingle) => m.mapping));
 
-    const addButton = withinModal.getByRole("button", {
-      name: /add/i,
+    await act(async () => {
```

Comment:
What does this `act` do?

Reply:
`act` is required to wait for all asynchronous changes in the component to be completed.

Reply:
From https://reactjs.org/docs/testing-recipes.html#act :
```
When writing UI tests, tasks like rendering, user events, or data fetching can be considered as “units” of interaction with a user interface. react-dom/test-utils provides a helper called [act()](https://reactjs.org/docs/test-utils.html#act) that makes sure all updates related to these “units” have been processed and applied to the DOM before you make any assertions.
```

I guess it would be best to wrap any "clicks" or renders with `act` for better consistency.

Reply:
I think anything that causes a re-render should be wrapped in `act`.

Reply:
`click` function provided by `react-testing-library` is already wrapped in `act`. Should all initial component renders be wrapped in `act`?

Reply:
Yes, I think it would be good to wrap them.